sdk/keyvault/Azure.Security.KeyVault.Certificates/src/CertificateOperation.cs

```diff
@@ -70,23 +86,17 @@ public override Response UpdateStatus(CancellationToken cancellationToken = defa
         {
             if (!_completed)
             {
-                Response<CertificateOperationProperties> pollResponse = _client.GetPendingCertificate(Properties.Name, _requestedOperation, cancellationToken);
+                Response<CertificateOperationProperties> pollResponse = _client.GetPendingCertificate(Properties.Name, cancellationToken);
 
                 _response = pollResponse.GetRawResponse();
 
                 Properties = pollResponse;
 
-                // Response value should only ever be null for a delete request.
+                // Properties will be null if deleted.
```

Comment:
Are you sure? Wouldn't it hit the https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/keyvault/Azure.Security.KeyVault.Shared/src/KeyVaultPipeline.cs#L228 and throw?

Reply:
`GetPendingCertificate` doesn't use that method. I added another set of methods to the `KeyVaultPipeline` to get the raw response so we can handle HTTP status codes without using exceptions for code flow:
```csharp
                Response response = await _pipeline.GetResponseAsync(RequestMethod.Get, cancellationToken, CertificatesPath, certificateName, "/pending").ConfigureAwait(false);
                switch (response.Status)
                {
                    case 200:
                    case 403:
                        return _pipeline.CreateResponse(response, new CertificateOperationProperties());

                    case 404:
                        return Response.FromValue<CertificateOperationProperties>(null, response);

                    default:
                        throw await response.CreateRequestFailedExceptionAsync().ConfigureAwait(false);
                }
```